src/grain/payouts.js

```diff
@@ -0,0 +1,180 @@
+ * to serious precision issues in practice.
+ */
+export const _multiply = (x: Grain, p: number) => {
+  const floated = Number(x) * p;
+  // $ExpectFlowError
+  return BigInt(Math.floor(floated));
+};
+
+/**
+ * Distribute an amount of Grain, strictly in proportion to the scores.
+ *
+ * This might be a good fit for distributing a fixed grain per week directly
+ * in proportion to the weekly cred score.
+ */
+export function fixedAmount(
+  harvestAmount: Grain,
+  scores: Map<NodeAddressT, number>
+): Map<NodeAddressT, Grain> {
+  if (harvestAmount < zero) {
+    throw new Error(`invalid harvestAmount: ${String(harvestAmount)}`);
+  }
+  if (scores.size === 0) {
+    return new Map();
+  }
+  const totalScore = sum(Array.from(scores.values()));
+  if (totalScore === 0) {
+    const perCapita = _multiply(harvestAmount, 1 / scores.size);
+    return MapUtil.mapValues(scores, () => perCapita);
+  }
+  const computeShare = (_, score) =>
+    _multiply(harvestAmount, score / totalScore);
+  return MapUtil.mapValues(scores, computeShare);
+}
+
+/**
+ * Distribute Grain so that each contributor's lifetime earnings
+ * are at least a fixed multiple of their score.
+ *
+ * Parameters:
+ *
+ * grainPerScore: Target amount of grain for each unit of score. This is
+ *   recorded as a number, and ignores grain precision (e.g. if
+ *   grainPerScore===2, then we are aiming to distribute two "full" grain per
+ *   unit of score, which might be tracked as 2n * 10n**18n).
+ *
+ * scores: The map from users' addresses to their scores. This could be
+ *   lifetime cred scores, it could be this week's scores, etc.
+ *
+ * earnings: The map from users' addresses to their lifetime distributed payout.
+ *   This information is used to determine which users have been "underpaid"
+ *   relative to the payout target, and by how much.
+ *
+ * Since this strategy provides a consistent reward per unit cred, it means that
+ * as the activity level (and thus, cred generation) in a project increases, the
+ * rewards also increase. This avoids the problem with fixed-reward strategies,
+ * where more participants mean less reward per person.
+ *
+ * If some users have been historically over-paid (e.g. because the
+ * target grainPerScore was higher in the past, or a weight change decreased
+ * their score), they might go for a long time without any payouts.
+ *
+ * Also, if users find a way to game/inflate cred scores, using this payout
+ * strategy will cause grain to inflate as well.
+ */
+export function fixedRatio(
+  grainPerScore: number,
+  scores: Map<NodeAddressT, number>,
+  earnings: Map<NodeAddressT, Grain>
+): Map<NodeAddressT, Grain> {
+  if (!isFinite(grainPerScore) || grainPerScore < 0) {
+    throw new Error(`invalid grainPerScore: ${grainPerScore}`);
+  }
+  return MapUtil.mapValues(scores, (addr, score) => {
+    const earned = earnings.get(addr) || zero;
+    const target = _multiply(one, grainPerScore * score);
```

Comment:
Would be nice if the `zero` and `one` constants were all caps for the sake of readability to make it clear that they are constants instead of parameters or variables
```suggestion
    const earned = earnings.get(addr) || ZERO;
    const target = _multiply(ONE, grainPerScore * score);
```

Reply:
Was resolved in https://github.com/sourcecred/sourcecred/pull/1710.